src/metax_api/tests/api/rest/base/views/datasets/write.py

```diff
@@ -3912,6 +3914,40 @@ def setUp(self):
         self._use_http_authorization(username=django_settings.API_EXT_USER['username'],
             password=django_settings.API_EXT_USER['password'])
 
+    def test_external_service_can_not_read_all_metadata_in_other_catalog(self):
+        ''' External service should get the same output from someone elses catalog than anonymous user '''
+        # create a catalog that does not belong to our external service
+        dc2 = DataCatalog.objects.get(pk=2)
+        dc2.catalog_json['identifier'] = 'Some other catalog'
+        dc2.catalog_record_services_read = 'metax'
+        dc2.force_save()
+
+        # Create a catalog record that belongs to some other user & our catalog nr2
+        cr = CatalogRecord.objects.get(pk=12)
+        cr.user_created = '#### Some owner who is not you ####'
+        cr.metadata_provider_user = '#### Some owner who is not you ####'
+        cr.data_catalog = dc2
+        cr.editor = None
+        cr.research_dataset['access_rights']['access_type']['identifier'] = ACCESS_TYPES['restricted']
+        cr.force_save()
+
+        # Let's try to return the data with our external services credentials
+        response_service_user = self.client.get('/rest/datasets/12')
+        self.assertEqual(response_service_user.status_code, status.HTTP_200_OK, response_service_user.data)
+
+        # Test access as unauthenticated user
+        self.client._credentials = {}
+        response_anonymous = self.client.get('/rest/datasets/12')
+        self.assertEqual(response_anonymous.status_code, status.HTTP_200_OK, response_anonymous.data)
+
+        self.assertEqual(response_anonymous.data, response_service_user.data,
+            "External service with no read-rights should not see any more metadata than anonymous user from a catalog")
+
+    def assert_catalog_record_not_open_access(self, cr):
+        from metax_api.models.catalog_record import ACCESS_TYPES
```

Comment:
This file has same imports in different tests as this one. I think they should be put on top with other imports.